packages/styletron-engine-atomic/src/cache.js

```diff
@@ -28,9 +28,13 @@ export class MultiCache<T> {
       cache.key = key;
       this.sortedCacheKeys.push(key);
       this.sortedCacheKeys.sort(sortMq);
-      const insertAtIndex = this.sortedCacheKeys.indexOf(key);
+      const keyIndex = this.sortedCacheKeys.indexOf(key);
+      const insertBeforeMedia =
+        keyIndex < this.sortedCacheKeys.length - 1
+          ? this.sortedCacheKeys[keyIndex + 1]
+          : "";
```

Comment:
Using `""` as the `insertBeforeMedia` value to trigger "append to container" logic seems a bit odd to me given that sheets with no media queries (i.e. `media === ""`) are ordered *before* sheets that do have media queries.

Perhaps it'd be better to use `undefined` instead to indicate the absence of a `insertBeforeMedia` value, which implies the sheet should not go before anything else and hence last.